backend/mongodb.js

```
@@ -34,59 +18,37 @@ const RETRY_DELAY_MS = 3000; // Wait 3 seconds between retries
  * Connect to MongoDB and initialize database references
  */
 async function connectToMongoDB() {
-  if (client) {
-    // Already connected
-    return { client, costDb, qtoDb };
+  if (client && client.topology && client.topology.isConnected()) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)